File: Python/array/degree_of_an_array.py
import collections
from typing import List

def findShortestSubArray(nums: List[int]) -> int:
    record = 0
    remain = collections.defaultdict(list)
    for i, num in enumerate(nums):
        if num in remain:
            remain[num][0] += 1
            remain[num][2] = i
        else:
            remain[num] = [1, i, i]
        record = max(record, remain[num][0])
    
    result = 50001
    for count, start, end in remain.values():
        if count == record:
            result = min(result, end - start + 1)
    
    return result

if __name__ == "__main__":
    print(findShortestSubArray([1]))

File: src/tmp/Java/src/com/dbc/algorithm/Array/DegreeOfAnArray.java
package com.dbc.algorithm.Array;

import java.util.HashMap;
import java.util.Map;

public class DegreeOfAnArray {
    public static int findShortestSubArray(int[] nums) {
        int record = 0, result = 50001;
        Map<Integer, int[]> remain = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (remain.containsKey(nums[i])) {
                remain.get(nums[i])[0]++;
                remain.get(nums[i])[2] = i;
            } else {
                remain.put(nums[i], new int[]{1, i, i});
            }
            record = Math.max(record, remain.get(nums[i])[0]);
        }
        for (Map.Entry<Integer, int[]> entry : remain.entrySet()) {
            if (record == entry.getValue()[0]) {
                result = Math.min(result, entry.getValue()[2] - entry.getValue()[1] + 1);
            }
        }
        return result;
    }
}
